core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -189,18 +196,21 @@ public MetadataResource putResource(final ServiceContext context, final String m
         // CMIS Secondary properties cannot be set in the same call that creates the document and sets it's properties,
         // so this is done in the following call after the document is created
         if (cmisConfiguration.existSecondaryProperty()) {
-            Map<String, Object> secondaryProperties = new HashMap<String, Object>();
+            Map<String, Object> secondaryProperties = new HashMap<>();
             setCmisMetadataUUIDSecondary(doc, secondaryProperties, metadataUuid);
             try {
                 doc.updateProperties(secondaryProperties);
+                if (MapUtils.isNotEmpty(additionalProperties)) {
+                    doc.updateProperties(additionalProperties,true);
+                }
```

Comment:
These two properties can be merged together by map.putAll(anotherMap). So the update property will happen after.

Reply:
You need to call setCmisMetadataUUIDSecondary after the putAll properties.  Otherwise the changes done from the setCmisMetadataUUIDSecondary will be overwritten by the putAll.

Reply:
Its been moved before the setCmisMetadataUUIDSecondary